src/jit/codegencommon.cpp

```diff
@@ -7319,14 +7321,39 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
         {
             noway_assert(varDsc->lvIsParam);
 
+            var_types storeType = varDsc->lvaArgType();
+
             if (!varDsc->lvIsRegArg)
             {
                 continue;
             }
 
-            var_types storeType = varDsc->lvaArgType();
-            regNumber argReg    = varDsc->lvArgReg;
-            getEmitter()->emitIns_S_R(ins_Store(storeType), emitTypeSize(storeType), argReg, varNum, 0);
+#if defined(UNIX_AMD64_ABI)
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+            if (varTypeIsStruct(varDsc))
+            {
+                SYSTEMV_AMD64_CORINFO_STRUCT_REG_PASSING_DESCRIPTOR structDesc;
+                CORINFO_CLASS_HANDLE typeHnd = varDsc->lvVerTypeInfo.GetClassHandle();
+                compiler->eeGetSystemVAmd64PassStructInRegisterDescriptor(typeHnd, &structDesc);
+                assert(structDesc.passedInRegisters);
+                unsigned __int8 offset0 = 0;
+                unsigned __int8 offset1 = 0;
+                var_types       type0   = TYP_UNKNOWN;
+                var_types       type1   = TYP_UNKNOWN;
+                compiler->GetStructTypeOffset(structDesc, &type0, &type1, &offset0, &offset1);
```

Comment:
I see this code got duplicated.  How about abstracting this in Compiler::GetStructTypeOffset() overload whose first argument is a CORINFO_CLASS_HANDLE and has out params to return offset and type of eightbytes of the struct?